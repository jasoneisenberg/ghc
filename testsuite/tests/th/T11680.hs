{-# LANGUAGE TemplateHaskell #-}
{-# LANGUAGE DuplicateRecordFields #-}


module T11680 where


import qualified Data.List as List


------------------------------------------------------------------------------
-- # No suggestions or exact match
ex1 :: Bool
ex1 = noMatches

------------------------------------------------------------------------------
-- # One suggestion (abcd), but no exact match
ex2 :: [a]
ex2 = abce

abcd :: String
abcd = "abcd"

------------------------------------------------------------------------------
-- # No suggestions, but an exact match after a splice
-- # Exact match following a single-line splice
-- # One splice separating use and binding
-- # Annotation does not affect identification of top-level splice
ex3 :: Int
ex3 = foo

------------------------------------------------------------------------------
-- # Suggestions (bat, baz) and an exact match after a splice
-- # Exact match following a multi-line splice
-- # Multiple splices separating use and binding
ex4 :: ()
ex4 = bar

bat :: [Double]
bat = [1.2, 3.4]

baz :: Bool
baz = True

------------------------------------------------------------------------------
-- # No suggestions, but an exact match IN a splice
ex5 :: [Double]
ex5 = ns

------------------------------------------------------------------------------
-- # Qualified suggestion and an exact local match after a splice
-- # Inline bracket does not affect identification of top-level splice
ex6 = intercalate

------------------------------------------------------------------------------
-- # Qualified suggestion, but no exact local match
ex7 = nub

------------------------------------------------------------------------------
-- # Nothing is reported for a duplicate field selector which is out-of-scope
ex8 :: Int
ex8 = x undefined


----------------------------------------------------------------------------
-- END OF INTER-SPLICE GROUP
----------------------------------------------------------------------------
$(return [])

{-# ANN foo (Just "hello") #-}
foo :: Int
foo = 23


----------------------------------------------------------------------------
-- END OF INTER-SPLICE GROUP
----------------------------------------------------------------------------
[d| ms = 3 : ns :: [Int]
    ns = 4 : ms
  |]

bar = [| 23.6 : bat |]

intercalate = undefined

data D = D { x :: Int
           , y :: Double
           }

data E = E { x :: Bool
           , y :: String
           }
