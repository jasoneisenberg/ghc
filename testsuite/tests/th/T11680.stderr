
T11680.hs:14:7: error: Variable not in scope: noMatches :: Bool

T11680.hs:19:7: error:
    • Variable not in scope: abce :: [a]
    • Perhaps you meant ‘abcd’ (line 22)

T11680.hs:30:7: error:
    • Variable not in scope: foo :: Int
    • ‘foo’ (line 72) is in scope after the splice on line 68

T11680.hs:37:7: error:
    • Variable not in scope: bar :: ()
    • ‘bar’ (line 82) is in scope after the splice on lines 78-80
      Perhaps you meant one of these: ‘bat’ (line 40), ‘baz’ (line 43)

T11680.hs:48:7: error:
    • Variable not in scope: ns :: [Double]
    • ‘ns’ is in scope beginning with the splice on lines 78-80

T11680.hs:53:7: error:
    • Variable not in scope: intercalate
    • ‘intercalate’ (line 84) is in scope
        after the splice on lines 78-80
      Perhaps you meant ‘List.intercalate’ (imported from Data.List)

T11680.hs:57:7: error:
    • Variable not in scope: nub
    • Perhaps you meant ‘List.nub’ (imported from Data.List)

T11680.hs:62:7: error: Variable not in scope: x :: t0 -> Int
